src/framework/MOM_diag_remap.F90

```diff
@@ -653,4 +660,33 @@ subroutine horizontally_average_diag_field(G, h, staggered_in_x, staggered_in_y,
 
 end subroutine horizontally_average_diag_field
 
+! Sets the source and target column thicknesses for reintegrate and remapping
+subroutine set_hsrc_hdest(is_zstar, hl_src, hr_src, hl_dest, hr_dest, h_src, h_dest)
+  logical,            intent(in   ) :: is_zstar !< Zstar is a special case
+  real, dimension(:), intent(in   ) :: hl_src   !< Left source column thicknesses
+  real, dimension(:), intent(in   ) :: hr_src   !< Right source column thicknesses
+  real, dimension(:), intent(in   ) :: hl_dest  !< Left source column thicknesses
+  real, dimension(:), intent(in   ) :: hr_dest  !< Right source column thicknesses
+  real, dimension(:), intent(  out) :: h_src    !< Source column for remapping/interpolation
+  real, dimension(:), intent(  out) :: h_dest   !< Targe column for remapping/interpolation
+
+  ! In the case of z-star coordinate, the u/v points should be associated with the
+  ! thinner T-point column
+  if (is_zstar) then
+    if ( SUM(hl_src)<SUM(hr_src) ) then
```

Comment:
This will do the wrong thing under an ice-shelf. @gustavo-marques will be unhappy with you and I'd keep your office mate happy if I were you. :)
For z* mode you'll need do loops like this
```
do k = 1:nk_src
  h_src(k) = min( hl_src(k), hr_src(k) )
enddo
do k = 1:nk_dest
  h_dest(k) = min( hl_dest(k), hr_dest(k) )
enddo
```